forwarder/forwarder/forwarderobject.py

```diff
@@ -151,10 +166,36 @@ def handle_app_update(self, task_id, future):
             print(f"*** TASK RETURN SUCCEEDED: {task_id} -- {time.time()}***")
             logger.info(f"Task:{task_id} succeeded")
 
-    def heartbeat_fail(self):
+    def _endpoint_heartbeat_fail(self):
         """Return true if too many heartbeats have been missed"""
         return int(time.time() - self.executor.last_response_time) > \
                     (self.max_heartbeats_missed * self.heartbeat_threshold)
+    
+    def _heartbeat_loop(self, kill_event):
+        while not kill_event.is_set():
+            logger.info("Activating executor.send_heartbeat()")
+            self.executor.send_heartbeat()
+            time.sleep(self.heartbeat_threshold)
```

Comment:
This is really bad naming. If we sleep for `heartbeat_threshold` we will miss the heartbeat. Let's call this `heartbeat_period`

Reply:
Agreed, and moreover, this will make the various modules have somewhat more consistent naming, though I think similar confusing names were used elsewhere in funcx